### PR TITLE
ACR base feature branch to create pipelines

### PR DESCRIPTION
**Pull request by seankane-msft**

### CODE REVIEW

**Comment by seankane-msft** [comment]
Getting ACR into a feature branch before working on auth flow. Next step is to create a live tests pipeline.

File: sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    def get_repository_client(self, repository, **kwargs):
+        # type: (str, Dict[str, Any]) -> ContainerRepositoryClient
+        """Get a repository client
+
+        :param repository: The repository to create a client for
+        :type repository: str
+        :returns: :class:~azure.containerregistry.ContainerRepositoryClient
+        """
+        return ContainerRepositoryClient(self.endpoint, repository, credential=self.credential, **kwargs)
```
Comment (kristapratico) [inline sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py:72]: I think we need to include a [transport wrapper](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/storage/azure-storage-blob/azure/storage/blob/_blob_service_client.py#L720) class here to ensure that in a situation like [this ](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/storage/azure-storage-blob/tests/test_common_blob.py#L2345)where context managers are used, the child client doesn't close the transport session of the parent client. Not blocking this PR, though

File: sdk/containerregistry/azure-containerregistry/azure/containerregistry/_models.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+class ContentPermissions(object):
+    def __init__(self, **kwargs):
+        self.delete_enabled = kwargs.get("delete_enabled")
+        self.list_enabled = kwargs.get("list_enabled")
+        self.read_enabled = kwargs.get("read_enabled")
+        self.write_enabled = kwargs.get("write_enabled")
+
+    @classmethod
+    def from_generated(cls, generated):
```
Comment (kristapratico) [inline sdk/containerregistry/azure-containerregistry/azure/containerregistry/_models.py:23]: should be private?

File: sdk/containerregistry/azure-containerregistry/azure/containerregistry/_models.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+            # registry=generated.registry,
+            manifest_count=generated.registry_artifact_count,
+            tag_count=generated.tag_count,
+            content_permissions=generated.writeable_properties,
+        )
+
+
+class RegistryArtifactOrderBy(int, Enum):
+
+    LastUpdateTimeDescending = 0
```
Comment (kristapratico) [inline sdk/containerregistry/azure-containerregistry/azure/containerregistry/_models.py:121]: Should use all caps - LAST_UPDATE_TIME_DESCENDING

File: sdk/containerregistry/azure-containerregistry/azure/containerregistry/_models.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    @classmethod
+    def from_generated(cls, generated):
+        return cls(
+            created_on=generated.created_on,
+            digest=generated.digest,
+            last_updated_on=generated.last_updated_on,
+            content_permissions=generated.modifiable_properties,
+            name=generated.name,
+            signed=generated.additional_properties.get("signed", None),
+            quarantine_state=generated.additional_properties.get("quarantineState", None),
```
Comment (kristapratico) [inline sdk/containerregistry/azure-containerregistry/azure/containerregistry/_models.py:170]: swagger need update?

File: sdk/containerregistry/azure-containerregistry/azure/containerregistry/_user_agent.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# coding=utf-8
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import sys
+
+from ._version import VERSION
+
+USER_AGENT = "azure-containerregistry/{}/{}".format(VERSION, sys.version)
```
Comment (kristapratico) [inline sdk/containerregistry/azure-containerregistry/azure/containerregistry/_user_agent.py:10]: I think azure-core will append the sys version for you

**Review by kristapratico — APPROVED** [review]
nits 🚀

File: sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py (hunk 1 above)
Comment (seankane-msft) [reply sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py]: Opening a tracking issue to address this before preview release. Thank you for pointing it out.

**Pull request closed without merging**